src/bonefish/messages/wamp_message.hpp

```diff
@@ -28,22 +29,47 @@ class wamp_message
 {
 public:
     wamp_message();
-    virtual ~wamp_message();
+    virtual ~wamp_message() = default;
 
-    wamp_message(const wamp_message& other) = delete;
+    wamp_message(const wamp_message&) = delete;
+    wamp_message(wamp_message&&) = delete;
     wamp_message& operator=(wamp_message const&) = delete;
+    wamp_message& operator=(wamp_message&&) = delete;
+
+    std::unique_ptr<msgpack::zone>&& release_zone();
+    const std::unique_ptr<msgpack::zone>& get_zone() const;
```

Comment:
So if I ditch unique_ptr would this actually then become:

msgpack::zone release_zone() { return std::move(m_zone); }
msgpack::zone& get_zone() { return m_zone; }